modules/exploits/linux/local/cve_2021_4034_pwnkit_lpe_pkexec.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Compile
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Local Privilege Escalation in polkits pkexec',
+        'Description' => %q{
+          A local privilege escalation vulnerability was found on polkit's pkexec utility.
+          The pkexec application is a setuid tool designed to allow unprivileged the
+          current version of pkexec doesn't handle the calling parameters count correctly
+          and ends trying to execute environment variables as commands.
+
+          An attacker can leverage this by crafting environment variables
+          in such a way it'll induce pkexec to execute arbitrary code. When successfully
+          executed the attack can cause a local privilege escalation given unprivileged
+          users administrative rights on the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Qualys Security', # Original vulnerability discovery
+          'Andris Raugulis', # Exploit writeup and PoC
+          'Dhiraj Mishra'    # Metasploit Module
+        ],
+        'DisclosureDate' => '2022-01-25',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2021-4034' ],
+          [ 'URL', 'https://www.qualys.com/2022/01/25/cve-2021-4034/pwnkit.txt' ],
+          [ 'URL', 'https://github.com/arthepsy/CVE-2021-4034' ] # PoC Reference
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ]
```

Comment:
Needs to add `ARTIFACTS_ON_DISK` since you're doing quite a bit with the filesystem.